src/NServiceBus.Core/Saga/Saga.cs

```diff
@@ -157,7 +157,20 @@ protected virtual void ReplyToOriginator(object message)
                 throw new Exception("Entity.Originator cannot be null. Perhaps the sender is a SendOnly endpoint.");
             }
 
-            Bus.Send(message, new SendOptions(Entity.Originator, Entity.OriginalMessageId));
+            var sendOptions = new SendOptions(correlationId: Entity.OriginalMessageId);
+            sendOptions.AsReplyTo(Entity.Originator);
+            sendOptions.ExtensionContext.Set(new WrapperForReplyToOriginator(true));
+            Bus.Send(message, sendOptions);
```

Comment:
Should we add a way to set the message intent explicitly here instead? (still a hack but less so IMO)